internal/controller/linodemachinetemplate_controller.go

```diff
@@ -141,9 +142,13 @@ func (lmtr *LinodeMachineTemplateReconciler) reconcile(ctx context.Context, lmtS
 		return ctrl.Result{}, nil
 	}
 
-	lmtScope.LinodeMachineTemplate.Status.Tags = slices.Clone(lmtScope.LinodeMachineTemplate.Spec.Template.Spec.Tags)
-
-	lmtr.Logger.Info("Successfully reconciled LinodeMachineTemplate", "name", lmtScope.LinodeMachineTemplate.Name)
+	// update the LMT status.tags if all the linodeMachines spec.tags is successfully updated.
+	if outErr == nil {
+		lmtScope.LinodeMachineTemplate.Status.Tags = slices.Clone(lmtScope.LinodeMachineTemplate.Spec.Template.Spec.Tags)
+		lmtr.Logger.Info("Successfully reconciled LinodeMachineTemplate", "name", lmtScope.LinodeMachineTemplate.Name)
+	} else {
+		lmtr.Logger.Error(outErr, "Error in reconciling LinodeMachineTemplate, retrying..", "name", lmtScope.LinodeMachineTemplate.Name)
```

Comment:
Retry won't happen unless returning something like `ctrl.Result{RequeueAfter: reconciler.DefaultLMTControllerReconcilerDelay}, nil`. I'm not sure there are any cases where we'd want to retry given we're just listing and patching machines.